docs/proposals/pepxxx_wheel_variant_support.md

```diff
@@ -62,88 +62,646 @@ something they [indirectly do today](https://github.com/wheelnext/wheelnext/pull
+last component in order to make it easy to distinguish different variants.
+
+The model defaults to using a hash in order to provide unique and reproducible filenames out of the box. However, it
+permits explicitly choosing a different label in order to make variants easy to recognize by humans. The label length
+is strictly limited in order to prevent the wheel filenames to become much longer than they are now, and causing issues
+on systems with smaller filename or path length limits.
+
+The same `-` character is used as the separator in order to reduce the risk of existing package manager implementations
+accidentally choosing a potentially unsupported wheel variant instead of a regular wheel. This was based on a survey
+of wheel filename verification methods used by different package managers and libraries (packaging, poetry, pip, uv).
+Both the current specification and some implementations are very permissive about different components, yet they all
+reject wheels if there are more than six components, or the build tag does not start with a digit. A limitation of this
+choice is that it assumes that the python tag will never start with a digit.
+
+#### Variant properties
+
+Variant properties follow a key-value design, where namespace and feature name constitute the key. Namespaces are used
+to group features defined by a single provider, and avoid conflicts should multiple providers define a feature with
+the same name. The character sets for all components are restricted in order to make it easier to preserve consistency
+between different providers, in particular uppercase characters are rejected to avoid different spellings of the same
+name. The character set for values is more relaxed, in order to permit values resembling versions and version
+specifications.
+
+Multiple values are permitted as a logical disjunction, while different features are treated conjunctively. This is
+meant to provide some flexibility in designating variant compatibility while avoiding having to implement a complete
+boolean logic. This flexibility is further extended via the concept of dynamic plugins, permitting the values to
+be dynamically interpreted, e.g. as version ranges.
+
+#### Variant hash
+
+Variant hash is used as a stable and unique identifier for every set of variant properties. It is truncated to
+8 characters in order to ensure that filenames remain short. SHA256 algorithm was chosen, because it is already widely
+used in wheels, in the `RECORD` file and therefore the package managers do not have to implement an additional
+algorithm.
+
+To ensure reproducible hash values, properties are sorted before hashing. They are then serialized into a canonical
+string form, and each one is terminated with a newline character to ensure their separation.
+
+As a special case, a variant hash of `00000000` is used for the null variant, in order to make it easily distinguishable
+from other variants.
+
+#### Null variant
```

Comment:
This section should probably say that publishing a null variant is optional, and that if it's missing then the non-variant-enabled wheel acts as the null variant.